pkg/docker/util.js

```diff
@@ -595,6 +595,62 @@
         return deferred.promise();
     };
 
+    util.delete_image_confirm = function confirm(client, image) {
+        var deferred = $.Deferred();
+        var image_id = image.image_id || image.Id;
+        var image_name = image.name || util.render_container_name(image.RepoTags[0]);
+
+        $('#delete-image-confirmation-dialog-title').text(
+            cockpit.format(_("Please confirm deletion of $0"), image_name));
+        $('#delete-image-confirmation-dialog-containers>table>tbody').empty();
+        client.containers_for_image(image_id).then(function(containers) {
+            var running_containers = [];
+
+            $(containers).each(function(index, value) {
+                var container = client.containers[value.Id];
+                var $row = $('<tr />')
+                    .append($('<td />').text(container.Name.startsWith('/') ?
+                                             container.Name.substring(1) : container.Name))
+                    .append($('<td />').text(util.render_container_status(container.State)));
+                $('#delete-image-confirmation-dialog-containers>table>tbody').append($row);
+                if (container.State.Running)
+                    running_containers.push(container.Id);
+            });
+
+            if (containers.length > 0) {
+                $('#delete-image-confirmation-dialog-body div:first-child').text(
+                    _("The following containers depend on this image and will become unusable."));
+                $('#delete-image-confirmation-dialog-containers').show();
+            } else {
+                $('#delete-image-confirmation-dialog-body div:first-child').empty();
+                $('#delete-image-confirmation-dialog-containers').hide();
+            }
+
+            if (running_containers.length > 0)
+                $('#delete-image-confirmation-dialog-confirm').text(_("Stop and delete"));
+            else
+                $('#delete-image-confirmation-dialog-confirm').text(_("Delete"));
+
+            function close() {
+                $('#delete-image-confirmation-dialog button').off('click');
+                $('#delete-image-confirmation-dialog').modal('hide');
+            }
+
+            $('#delete-image-confirmation-dialog-confirm').click(function () {
+                close();
+                deferred.resolve(running_containers, containers.length > 0);
+            });
+
+            $('#delete-image-confirmation-dialog-cancel').click(function () {
```

Comment:
There are a lot of `$('#delete-image-confirmation-dialog-cancel')` references in the JS. I've found it better to put an element in a variable like `$delete_image_confirmation_dialog` and then reference that variable... otherwise the code is doing function calls and element lookups every time the element is referenced. It has the added bonus of having better compression in the cases where JS code is minimized too, as it's a bunch of variables instead of strings in functions.

(With jQuery elements, I usually give them a variable name that starts with `$`. However, I'm not sure what Cockpit does with regard to this.)

For things like `$('#delete-image-confirmation-dialog button')`, it's possible to do either `$delete_image_confirmation_dialog.find('button')` or `$('button', $delete_image_confirmation_dialog)` (or whatever the variable is called).